harness/suite.go

```diff
@@ -280,3 +294,43 @@ func (s *Suite) runTests(w io.Writer) error {
 func (s *Suite) outputPath(path string) string {
 	return filepath.Join(s.opts.OutputDir, path)
 }
+
+// cleanOutputDir creates/empties Options.OutputDir.
+// If the path already exists it must be named similar to `_foo_temp`
+// or contain `.harness_temp` to indicate removal is safe; we don't
+// want users wrecking things by accident with `-outputdir /tmp`
+func (s *Suite) cleanOutputDir() error {
```

Comment:
Not sure if this is the right place, but it might be worth surfacing an error that, based on the user given name, the output dir wasn't safe to remove.